### PR TITLE
TSK-1982 refactore QueryClassificationWithPaginationAccTest to use Test-API

### PR DESCRIPTION
**Pull request by arolfes**
[SonarCloud](https://sonarcloud.io/summary/new_code?id=arolfes_taskana&branch=TSK-1982-refactore-QueryClassificationWithPaginationAccTest)

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  @Test
+  void should_GetFirstFiveClassificationsOnAPage_When_FirstPageAndPageSizeIsFive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").listPage(1, 5);
+    assertThat(results).hasSize(5);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetLastFourClassificationsOnAPage_When_FourthPageAndPageSizeIsFive() {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:87]: What do you mean with "PageSize" in the test name? Is it the same as "Limit" in previous test names? If I understand this test correctly, the size of the last page is four. That's why the test name is confusing to me

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  @Test
+  void should_GetZeroClassificationsOnAPage_When_PageSizeIsNegative() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").listPage(1, -1);
+    assertThat(results).hasSize(0);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetClassificationsOnAPage_When_PagenumberIsZeroButPageSizeIsPositive() {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:119]: What is the difference to "should_GetFirstFiveClassifications_When_OffsetIsZeroAndLimitIsFive" ? Do we need this test?

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").listPage(0, 1);
+    assertThat(results).hasSize(1);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetClassificationsOnAPage_When_PagenumberIsNegativeButPageSizeIsPositive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").listPage(-1, 1);
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:129]: What is the difference to "should_GetOneClassifications_When_OffsetIsNegativeAndLimitIsPositive" ?

**Review by ryzheboka — CHANGES_REQUESTED** [review]

**Comment by ryzheboka** [comment]
Hi @arolfes , if you don't have time to finish this PR, we could make the changes :)

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java (hunk 1 above)
Comment (jamesrdi) [reply lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:87]: The size of last page is four because there are not enough tasks

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java (hunk 2 above)
Comment (jamesrdi) [reply lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:119]: This Test tests the pagination function listPage(), but the other does not test the pagination

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+import org.junit.jupiter.api.Test;
+
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.models.ClassificationSummary;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.security.WithAccessId;
+
+@TaskanaIntegrationTest
+public class ClassificationQueryPaginationAccTest {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:17]: Rename to "ClassificationQueryOffsetAndPaginationAccTest"

File: lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  @Test
+  void should_GetOneClassifications_When_OffsetIsNegativeAndLimitIsPositive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(-1, 1);
+    assertThat(results).hasSize(1);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetFirstFiveClassificationsOnAPage_When_FirstPageAndPageSizeIsFive() {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java:79]: Please include "WithPagination" in the test name. For example. "_GetFirstFiveClassifications_WhenUsingPaginationWithPageOneAndSizeFive_" Please rename other tests with pagination accordingly

**Review by ryzheboka — CHANGES_REQUESTED** [review]

**Review by jamesrdi — APPROVED** [review]

**Review by ryzheboka — APPROVED** [review]
